aws/services/cw/policy.ftl

```diff
@@ -17,7 +17,18 @@
                     "logs:DescribeLogStreams"
                 ],
                 logGroupArn)
-        ]
+        ] +
+        (osFamily = "windows")?then(
+            [
+                getPolicyStatement(
+                    [
+                        "cloudwatch:PutMetricData",
+                        "ec2:DescribeTags"
+                    ],
+                    "*")
+            ],
+            []
+        )
```

Comment:
I don't think these are OS dependent.  the linux instances can put metric data as well. We just didn't have it setup. Probably better to make this standard permissions on cwLogs across the board

Reply:
Cool, was just trying to constrain my changes to avoid breaking something out of ignorance

Reply:
Its just that this update isn't conditional on if you are using windows or not. Its more conditional on if you are using metrics as part of your cloudwatch agent configuration. Along with that we probably need to apply some conditions to the PutMetricData action as it's a write action. If I had multiple products in this account this permission could put data to any metric

Reply:
I hadn't considered products. Logically I should be restricting it base on product and environment, but I don't see how to set that through the permissions ARN. Can you point me in the right direction?

Reply:
This is the one you want 
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/iam-cw-condition-keys-namespace.html 

So either there is a namespace created for each product/env/segment or we limit it to a single namespace where all ec2 instances log their data under different dimensions

Reply:
I think the piece of the puzzle that I'm missing is hidden in that last sentence. As product/env/segment are constructs of ours, how do they relate to ARNs (I'm assuming an ARN is a unique reference, correct?)

Reply:
So its within the configuration of the agent. 

https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/CloudWatch-Agent-Configuration-File-Details.html 

within the Metrics configuration section there is the definiton of the namespace. This is the top level grouping that is used to store CloudWatch Metrics under 

So we could create a namespace that is based on the name of the component or the path of the component name. Then all instance metrics would be stored under that using the instance Id as a dimension which allows for metrics to be stored under different granularities 

Then when creating the IAM policy you would provide the Metric namespace that the role has permissions to write into

Reply:
Cool, thanks, done